### PR TITLE
Complete Day 6: Scaling Implementation + Developer Ergonomics (Part 1)

### PR DESCRIPTION
**Pull request by jeffreyhorn**
Implemented Curtis-Reid and byvar scaling algorithms with CLI integration.

Features Added:
- Curtis-Reid scaling algorithm with iterative row/column norm balancing
- Byvar (per-variable) scaling mode for column-only normalization
- --scale CLI flag (none|auto|byvar, default: none)
- Scaling factor computation and storage in KKT system
- Enhanced parser error messages with source locations

Implementation Details:
- Curtis-Reid: Geometric mean scaling R[i,i]=1/√(row_norm_i), C[j,j]=1/√(col_norm_j)
- Structural scaling: Uses 1.0 for all nonzero entries (symbolic Jacobian)
- Byvar: Column-only normalization, no row scaling
- Integration: Compute scaling after derivatives, store in KKT for future emission

Files Modified:
- src/kkt/scaling.py (NEW, 191 lines): Curtis-Reid and byvar algorithms
- src/cli.py (+47 lines): --scale flag, scaling computation integration
- src/config.py (+8 lines): scale parameter validation
- src/kkt/kkt_system.py (+5 lines): scaling factor storage
- src/ir/parser.py (+12 lines): enhanced error messages

Tests Added:
- tests/unit/kkt/test_scaling.py (247 lines, 14 tests)
  * 6 Curtis-Reid tests (simple, conditioning, convergence, edge cases)
  * 4 byvar tests (simple, badly scaled, empty columns)
  * 4 edge case tests (single row/col, empty, large matrices)

Acceptance Criteria Met (10/10):
- [x] Curtis-Reid scaling with iterative row/col balancing
- [x] Byvar mode for per-variable scaling
- [x] Scaling from symbolic Jacobian (structural, 1.0 for nonzeros)
- [x] Balanced row/col norms verified in tests
- [x] --scale none (default, backward compatible)
- [x] --scale auto (Curtis-Reid)
- [x] --scale byvar (column-only)
- [x] All 779 original tests pass with default
- [x] 14 new tests verify semantic preservation
- [x] Parser errors include source locations

Test Results: 793 passed, 1 skipped, 1 xfailed
- 779 original tests (unchanged)
- 14 new scaling tests
- No regressions

Quality Checks:
✅ make typecheck: 43 source files, 0 errors
✅ make lint: All checks passed
✅ make format: 107 files unchanged
✅ make test: 793 passed

Updated Documentation:
- PLAN.md: All Day 6 acceptance criteria checked off
- README.md: Day 6 marked complete
- CHANGELOG.md: Comprehensive Day 6 summary with technical details

Future Work:
- Day 7: Matrix Market export, model statistics, logging
- Value-based scaling (evaluate expressions at a point)
- GAMS emission of scaling factors
- PATH solver validation with scaling

### CODE REVIEW

File: src/kkt/scaling.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        col_norms = np.where(col_norms > min_norm, col_norms, 1.0)
+        C_k = 1.0 / np.sqrt(col_norms)
+
+        # Apply column scaling: each column j is multiplied by C_k[j]
+        dense_jac = dense_jac * C_k[np.newaxis, :]
+        C = C_k * C  # Accumulate scaling
+
+        # Check convergence
+        max_row_dev = np.abs(row_norms - 1.0).max()
+        max_col_dev = np.abs(col_norms - 1.0).max()
```
[the remark this answers is not in the document]
Comment (jeffreyhorn) [reply src/kkt/scaling.py]: Issue**: The convergence check was using pre-scaling norms instead of post-scaling norms, which could incorrectly report convergence.

**Fix Applied** (`src/kkt/scaling.py`):
- Added recomputation of norms after both row and column scaling
- Now checks `row_norms_post` and `col_norms_post` to verify the scaled matrix actually has balanced norms ≈ 1.0
- This ensures accurate convergence detection

**Code Change**:
```python
# Before: Checked pre-scaling norms (incorrect)
max_row_dev = np.abs(row_norms - 1.0).max()
max_col_dev = np.abs(col_norms - 1.0).max()

# After: Check post-scaling norms (correct)
row_norms_post = np.linalg.norm(dense_jac, axis=1, ord=2)
col_norms_post = np.linalg.norm(dense_jac, axis=0, ord=2)
max_row_dev = np.abs(row_norms_post - 1.0).max()
max_col_dev = np.abs(col_norms_post - 1.0).max()
